### PR TITLE
manager/integration/tests: add test skeleton to test data locality feature

### PR DESCRIPTION
**Pull request by PhanLe1010**
longhorn/longhorn#1045

### CODE REVIEW

File: manager/integration/tests/test_scheduling.py [hunk 1]
```diff
@@ -377,3 +377,50 @@ def test_replica_rebuild_per_volume_limit():
...
+       a replica on the new-engine-node and remove the replica on
+       old-engine-node
+    10. Update dataLocality to disabled for volume(1)
+    11. detach the volume(1) and attach it to a different node.
+       Let's call the new node is new-engine-node and the old
+       node is old-engine-node
+    12. Wait for volume(1) to finish detaching
+    13. Use a retry loop to verify that Longhorn does not create
+        a replica on the new-engine-node
+    14. clean up
```
Comment (yasker) [inline manager/integration/tests/test_scheduling.py:424]: What about the case that replica cannot be created on the attached node?

Also, what if the volume detached (workload shutdown) during the time of detaching? Need to think more about negative cases.

Comment (PhanLe1010) [reply manager/integration/tests/test_scheduling.py:424]: Thank you! I will fix this

Comment (PhanLe1010) [reply manager/integration/tests/test_scheduling.py]: @yasker I added the 2 negative test cases and also check for replica deletion preference.

**Review by meldafrawi — APPROVED** [review]

**Comment by meldafrawi** [comment]
LGTM

**Pull request merged**